### PR TITLE
fix(newrelic): handle invalid newrelic env vars correctly and add tests

### PR DESCRIPTION
**Pull request by ardatan**
I'll add tests for resolver tracking later

### CODE REVIEW

**Review by saihaj — APPROVED** [review]

**Comment by saihaj** [comment]
Is there a way we can add tests for this?

**Comment by ardatan** [comment]
@saihaj I think so;
https://github.com/newrelic/node-test-utilities
Hmm maybe next sprint I can take a look at this.

File: packages/plugins/newrelic/src/index.ts [hunk 1]
```diff
@@ -125,50 +119,75 @@ export const useNewRelic = (rawOptions?: UseNewRelicOptions): Plugin => {
...
+          ) as FieldNode[];
+          rootFields = fieldNodes.map(fieldNode => fieldNode.name.value);
+        }
+
+        const operationType = rootOperation.operation;
+        const operationName =
+          options.extractOperationName?.(args.contextValue) ||
+          args.operationName ||
+          rootOperation.name?.value ||
+          AttributeName.ANONYMOUS_OPERATION;
```
Comment (YassinEldeeb) [inline packages/plugins/newrelic/src/index.ts]: I think this variable has been defined at the top with the same definition and is accessible from line 140
```suggestion
```

**Review by YassinEldeeb — CHANGES_REQUESTED** [review]
LGTM, besides removing the repeated `operationName` variable

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -71,6 +71,8 @@ jobs:
         run: yarn test
         env:
           CI: true
+      - name: Test NewRelic
```
Comment (saihaj) [inline .github/workflows/tests.yml]: Why add a new command. Our CI run tests recursively for each package?

Comment (ardatan) [reply .github/workflows/tests.yml]: No, `yarn test` runs the `test` script defined in the root `package.json`. And that script runs `jest` only.

**Pull request merged**